### ISSUE TITLE
AOSP v400[a-c]: Camera and Flashlight doesn't work on camera2api with LEGACY support

### ISSUE THREAD

**Issue by thongass000** [phhusson/treble_experimentations#2071]
The problem is simple. So my device is Samsung A9 2018, with sdm600 chipset. The device's camera apparently only works with camera2api with LEGACY support (if I force enabling camera2api HAL3, which reaches FULL support level, all camera apps will crash, even on stock firmware).
Trying all Android 12 builds so far, I've been dealing with the problem of unusable flashlight and camera and the missing of the legacy camera app (installing 3rd-party camera apps will notify that the camera is being used by another app). After enabling FULL support by editing system/build.prop, I've found that the flashlight and the legacy camera app are back, but, as I have mentioned, the camera won't work.
I hope that this issue will be fixed soon

**Comment by phhusson** [comment]
Yup, Android 12 removed support for Camera HAL 1//Legacy Bringing it back
will probably require a lot of work.

Le jeu. 11 nov. 2021 à 13:30, James Nguyen ***@***.***> a
écrit :

**Comment by ponces** [comment]
A user has reported that forcing HAL3 and going permissive has fixed their issues. Is it possible that this is just a matter of fixing some selinux stuff?
